e2e/test_operator/operator_test.go

```diff
@@ -1976,4 +1976,71 @@ var _ = Describe("Operator", Label("e2e", "pr"), func() {
 			})
 		})
 	})
+
+	When("maxConcurrentReplacements is lower than the number of storage pods", func() {
+		var initialConcurrentReplacements *int
+		var initialPodUpdateStrategy fdbv1beta2.PodUpdateStrategy
+		var initialReplaceInstancesWhenResourcesChange *bool
+
+		BeforeEach(func() {
+			// Remember the current settings before updating the spec
+			initialConcurrentReplacements = fdbCluster.GetCluster().Spec.AutomationOptions.MaxConcurrentReplacements
+			initialPodUpdateStrategy = fdbCluster.GetClusterSpec().AutomationOptions.PodUpdateStrategy
+			initialReplaceInstancesWhenResourcesChange = fdbCluster.GetCluster().Spec.ReplaceInstancesWhenResourcesChange
+			// Allow to replacement of 3 pods concurrently, as there are 5 storage servers, there need to be at least 2 rounds of replacements to replace all.
+			spec := fdbCluster.GetCluster().Spec.DeepCopy()
+			spec.AutomationOptions.MaxConcurrentReplacements = pointer.Int(3)
+			spec.AutomationOptions.PodUpdateStrategy = fdbv1beta2.PodUpdateStrategyDelete
+			spec.ReplaceInstancesWhenResourcesChange = pointer.Bool(true)
+			fdbCluster.UpdateClusterSpecWithSpec(spec)
+		})
+
+		AfterEach(func() {
+			// Reset to the initial settings
+			spec := fdbCluster.GetCluster().Spec.DeepCopy()
+			spec.AutomationOptions.MaxConcurrentReplacements = initialConcurrentReplacements
+			spec.AutomationOptions.PodUpdateStrategy = initialPodUpdateStrategy
+			spec.ReplaceInstancesWhenResourcesChange = initialReplaceInstancesWhenResourcesChange
+			fdbCluster.UpdateClusterSpecWithSpec(spec)
+		})
+
+		When("a change that requires a replacement of all storage pods", func() {
+			var initialVolumeClaims *corev1.PersistentVolumeClaimList
+			var newCPURequest, initialCPURequest resource.Quantity
+			BeforeEach(func() {
+
+				initialVolumeClaims = fdbCluster.GetVolumeClaimsForProcesses(fdbv1beta2.ProcessClassStorage)
```

Comment:
```suggestion
			BeforeEach(func() {
				initialVolumeClaims = fdbCluster.GetVolumeClaimsForProcesses(fdbv1beta2.ProcessClassStorage)
```

It's probably better to store the UID in a slice, that makes the comparison easier and also will prevent cases where the comparison is failing because of a different pointer.